patrikellboj/alm02-patrike-marios-victorj-marcuss
# Generating test reports

Note!
=======
This pull request is building on previous commits in other branches such as "dockerAndJenkinsTest" and "controller".

Problem
=======
No test report or test coverage are being published when running pipeline.
No .jar file of project is being archived

Solution
========
Added script in Jenkinsfile that creates shurfire unit test report and jacoco test coverage report. Necessary plugins also added to pom.xml file.
Also added a script for archiving the generated .jar file on success.

Change summary:
---------------
* Script in Jenkinsfile that can be run through pipeline
* Necessary plugins added to pom.xml
* Script for archiving generated .jar file

Steps to Verify:
----------------
1. See if you can set up a pipeline and get test reports (surefire unit test report and jacoco test coverage) and also the archived .jar file
2. Download .jar file and run "java -jar exampleJarFile.jar" in terminal to see if you can access the project on port localhost:8085
3. Terminate program through "ctrl + c"

Screenshots:
-----------------------
<img width="1436" alt="test-reports" src="https://user-images.githubusercontent.com/32056521/92304821-bbdf1380-ef81-11ea-921c-fc26f2409302.png">
